src/PodCreateModal.jsx

```diff
@@ -76,33 +92,68 @@ export const PodCreateModal = ({ user, systemServiceAvailable, userServiceAvaila
     };
 
     const onCreateClicked = () => {
+        if (!validateForm())
+            return;
         const createConfig = getCreateConfig();
         createPod(owner === systemOwner, createConfig);
     };
 
-    const onValueChanged = (key, value) => {
-        if (key === "podName") {
-            setPodName(value);
-        }
-        if (utils.is_valid_container_name(value)) {
-            setNameError(null);
-        } else {
-            setNameError(_("Invalid characters. Name can only contain letters, numbers, and certain punctuation (_ . -)."));
-        }
+    const isFormInvalid = validationFailed => {
+        const groupHasError = row => Object.values(row)
+                .filter(val => val) // Filter out empty/undefined properties
+                .length > 0; // If one field has error, the whole group (dynamicList) is invalid
+
+        // If at least one group is invalid, then the whole form is invalid
+        return validationFailed.publish?.some(groupHasError) ||
+            !!validationFailed.podName;
+    };
+
+    const validatePodName = value => {
+        if (!utils.is_valid_container_name(value))
+            return _("Invalid characters. Name can only contain letters, numbers, and certain punctuation (_ . -).");
+    };
+
+    const validateForm = () => {
+        const newValidationFailed = { };
+
+        const publishValidation = publish.map(a => {
+            return {
+                IP: validatePublishPort(a.IP, "IP"),
+                hostPort: validatePublishPort(a.hostPort, "hostPort"),
+                containerPort: validatePublishPort(a.containerPort, "containerPort"),
+            };
+        });
+        if (publishValidation.some(entry => Object.keys(entry).length > 0))
+            newValidationFailed.publish = publishValidation;
+
+        const podNameValidation = validatePodName(podName);
+
+        if (podNameValidation)
+            newValidationFailed.containerName = podNameValidation;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1482-20231110-112222-2df94782-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)